macOS/unwindmacos.cpp

```diff
@@ -388,7 +407,11 @@ bool UnwindGeneratorMacOS::HandleBreakpoint(ModuleInfo* module, void *address) {
 
       auto it = unwind_data->return_addresses.find(ip);
       if (it != unwind_data->return_addresses.end()) {
+#ifdef ARM64
+        ip = it->second.original_return_address;
```

Comment:
The reason for -1 in Intel case is this piece of code (and relevant comment) from UnwindCursor::setInfoBasedOnIPRegister

```
  // If the last line of a function is a &quot;throw&quot; the compiler sometimes
  // emits no instructions after the call to __cxa_throw.  This means
  // the return address is actually the start of the next function.
  // To disambiguate this, back up the pc when we know it is a return
  // address.
  if (isReturnAddress)
    --pc;
```

On Intel isReturnAddress ende up being `false` because __Unwind_SetIP always sets this argument to `false` so this branch isn't taken even though we are dealing with return addresses. However, since on arm we call setInfoBasedOnIPRegister directly and IP isn't decremented here, we should probably (?) call `setInfoBasedOnIPRegister(true)` instead of `setInfoBasedOnIPRegister(false)`

Reply:
done.